src/Form/Label.js

```diff
@@ -9,7 +9,6 @@ const Label = createComponent({
     transition: 250ms;
     font-weight: 500;
     margin: 0 0 4px 4px;
-    z-index: 10;
```

Comment:
@schlegz noted that dropdowns and floating input labels have competing z-indices:

<img width="977" alt="Screen Shot 2019-07-31 at 12 24 45 PM" src="https://user-images.githubusercontent.com/6404663/62243357-fbf0c180-b391-11e9-85bb-254ef8074a00.png">

Removing the z-index still achieves the same visual appearance, and resolves the conflict.
![image](https://user-images.githubusercontent.com/16051172/62249271-d9b17080-b39e-11e9-923d-f19c63876ecc.png)


Are there any use cases for this z-index? Do we still want to keep it, but just bump it down to something less than 10?